app/src/main/java/com/checkin/app/checkin/auth/activities/AuthenticationActivity.kt

```diff
@@ -0,0 +1,290 @@
+                        authViewModel.isLoginAttempt = false
+                        if (!::phoneNo.isInitialized) {
+                            toast("Invalid Mobile Number")
+                            pass
+                        }
+                        askUserDetails()
+                    } else {
+                        val error = resource.errorBody ?: return@Observer
+                        when {
+                            error.has("errors") -> {
+                                val msg = error["errors"][0].asText()
+                                toast(String.format(Locale.ENGLISH, "%s\nTry again.", msg))
+                            }
+                            error.has("username") -> {
+                                authViewModel.showError(error)
+
+                            }
+                            resource.message != null -> toast(resource.message)
+                        }
+                    }
+                    else -> {
+                        hideProgress()
+                        Utils.toast(this, resource.message)
+                    }
+                }
+            }
+        })
+
+        tvTermsAndConditions.apply {
+            text = Utils.fromHtml(getString(R.string.terms_and_condition_underlined))
+            setOnClickListener { startActivity(Intent(Intent.ACTION_VIEW, getString(R.string.url_app_terms_conditions).toUri())) }
+        }
+    }
+
+    private fun authenticateWithCredential(credential: AuthCredential) {
+        firebaseAuth.signInWithCredential(credential)
+                .addOnCompleteListener(this) { task: Task<AuthResult?> ->
+                    if (task.isSuccessful) {
+                        assert(firebaseAuth.currentUser != null)
+                        firebaseAuth.currentUser!!.getIdToken(false).addOnSuccessListener { tokenResult: GetTokenResult ->
+                            authViewModel.setFireBaseIdToken(tokenResult.token)
+                            authViewModel.login()
+                        }
+                    } else if (task.exception != null) {
+                        Utils.logErrors(TAG, task.exception, "Authentication failed")
+                        Toast.makeText(applicationContext, R.string.error_authentication, Toast.LENGTH_SHORT).show()
+                    }
+                }
+    }
+
+
+    private fun askUserDetails() {
+        val user = firebaseAuth.currentUser
+        if (user == null) {
+            Log.e(TAG, "Logged-in user is NULL")
+            return
+        }
+        val name = user.displayName
+        goBack = false
+
+        val fullNameSeparatedList = name?.split("\\s+".toRegex()) ?: listOf(defaultFirstName)
+        //create an user
+        onUserInfoProcess(firstName = fullNameSeparatedList[0], lastName = if (fullNameSeparatedList.size > 1) fullNameSeparatedList[1]
+        else null, username = phoneNo, gender = defaultGender)
+
+        authViewModel.authResult.observe(this, Observer {
```

Comment:
I'm not sure why you're trying to add another observer here... since all the observers would be called and you're adding it everytime on call of _askUserDetails_